src/fhir.js

```diff
@@ -301,15 +391,74 @@ class Patient extends FHIRObject {
       return [];
     }
     const resourceType = classInfo.name.replace(/^FHIR\./, '');
-    const records = this._bundle.entry.filter( e => {
-      return e.resource && e.resource.resourceType == resourceType;
-    }).map( e => {
-      return new FHIRObject(e.resource, classInfo, this._modelInfo);
-    });
+    const records = this._bundle.entry
+      .filter(e => {
+        return e.resource && e.resource.resourceType == resourceType;
+      })
+      .map(e => {
+        return new FHIRObject(e.resource, classInfo, this._modelInfo);
+      });
     return records;
   }
 }
 
+class AsyncPatient extends FHIRObject {
+  constructor(patientData, modelInfo, serverData) {
+    const patientClass = modelInfo.patientClassIdentifier
+      ? modelInfo.patientClassIdentifier
+      : modelInfo.patientClassName;
+    const ptClass = modelInfo.findClass(patientClass);
+    super(patientData, ptClass, modelInfo);
+
+    // Define a "private" un-enumerable property to hold the patient data
+    Object.defineProperty(this, '_patientData', {
+      value: patientData,
+      enumerable: false
+    });
+    Object.defineProperty(this, '_serverData', {
+      value: serverData,
+      enumerable: false
+    });
+  }
+
+  async findRecords(profile) {
+    const classInfo = this._modelInfo.findClass(profile);
+    if (classInfo == null) {
+      console.error(`Failed to find type info for ${profile}`);
+      return [];
+    }
+    const resourceType = classInfo.name.replace(/^FHIR\./, '');
+    // If the patient resource type is requested, return array with just this resource
+    if (resourceType === 'Patient') {
+      return [this];
+    }
+
+    const compartmentInfo = patientCompartmentDefinition.resource.filter(
+      def => def.code === resourceType
+    );
+    if (!compartmentInfo[0].param) {
+      console.error(`Resource type: ${resourceType} cannot reference a patient.`);
```

Comment:
```suggestion
      throw new Error(`Resource type: ${resourceType} cannot reference a patient.`);
```

Reply:
I tried removing entries from the compartment definition and when I did, I received a TypeError for not being able to read 'param' of undefined in line 439 since `compartmentInfo` became an empty array. Similarly, when `compartmentInfo` was empty and the error got logged, I got the same TypeError later on, so I think it makes sense to instead check that an entry exists in the array and then throw an Error instead of using `console.error()`